plus/src/test/java/org/apache/calcite/slt/TestCalcite.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.slt;
+
+import org.apache.calcite.slt.executors.CalciteExecutor;
+
+import net.hydromatic.sqllogictest.OptionsParser;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+/**
+ * Tests using sql-logic-test suite.
+ */
+public class TestCalcite {
```

Comment:
The class should be reworked. 

1. We want to run all sql-logic-test that pass and not only `select1.test`. 
2. When there is a regression we want to have a clear way to know which test and which file failed. (optional but useful)
3. We only care about tests using the Calcite executor; anything else is out of context of this pr.

Reply:
The main problem is that there are *lots* of failures. Only in these 1000 tests from select1.test there are 153 failures. They are due to a small number of underlying causes. If we run all 5M tests in this call, with this expected failure rate there will be 50K failures, and the output will cause the PrintStreams used in launchSqlLogicTest to run out of memory.

Until we implement https://github.com/hydromatic/sql-logic-test/issues/16 we probably cannot run all the tests.

When all the found Calcite bugs are fixed we can modify the test here to assert that there are no failures, but until that happens I am unsure how to use this test. Right now this code is most useful if it outputs a diagnosis for human consumption.

Reply:
There is nothing preventing us from re-redirecting the streams to files and not keep them in byte buffers. Hopefully by doing this we should avoid running out of memory.

We don't necessarily need to assert that there are no failures but more like that the number of failures doesn't change. Unless the number of failures goes down which I guess is a good thing meaning that we fixed a bug.

As far as I understand, point 2 that I suggested above is hard to implement at this stage given that there are no test files that pass completely so we can definitely defer to another JIRA.